pkg/registry/core/service/portallocator/controller/repair.go

```diff
@@ -115,9 +124,8 @@ func (c *Repair) runOnce() error {
 
 	rebuilt := portallocator.NewPortAllocator(c.portRange)
 	// Check every Service's ports, and rebuild the state as we think it should be.
-	for i := range list.Items {
-		svc := &list.Items[i]
-		ports := service.CollectServiceNodePorts(svc)
+	for _, svc := range list.Items {
```

Comment:
Please revert - this was done on purpose to avoid unnecessary copies for performance reasons.